src/test/java/org/owasp/esapi/configuration/EsapiPropertyManagerTest.java

```diff
@@ -304,6 +344,44 @@ public void testBooleanPropertyNotFoundByLoaderAndThrowException() {
         // then expect exception
     }
 
+    @Test
+    public void testBooleanPropertyNotFoundByLoaderWithDefaultValueTrue() {
+        // given
+        String propertyKey = "non.existing.property";
+        boolean expectedPropertyValue = true;
+
+        // when
+        try {
+            testPropertyManager = new EsapiPropertyManager();
+        } catch (IOException e) {
+            fail(e.getMessage());
+        }
+
+        boolean propertyValue = testPropertyManager.getBooleanProp(propertyKey, true);
+
+        // then
+        assertEquals(expectedPropertyValue, propertyValue);
+    }
+
+    @Test
+    public void testBooleanPropertyNotFoundByLoaderWithDefaultValueFalse() {
+        // given
+        String propertyKey = "non.existing.property";
+        boolean expectedPropertyValue = false;
+
+        // when
+        try {
+            testPropertyManager = new EsapiPropertyManager();
+        } catch (IOException e) {
+            fail(e.getMessage());
+        }
+
+        boolean propertyValue = testPropertyManager.getBooleanProp(propertyKey, false);
+
+        // then
+        assertEquals(expectedPropertyValue, propertyValue);
+    }
+
```

Comment:
If we decide to revise this behavior to throw a `ConfigurationException` if `getBooleanProperty(String, Boolean)` is called and a non-Boolean property if found, we will need a new test here.

Ditto in **StandardEsapiPropertyLoaderTest.java** and **XmlEsapiProertperLoaderTest.java**.